stumpy/aamp_motifs.py

```diff
@@ -398,11 +401,14 @@ def max_distance(D):
     else:
         candidate_idx = np.argmin(D)
 
-    while (
-        D[candidate_idx] <= atol + max_distance
-        and np.isfinite(D[candidate_idx])
-        and len(matches) < max_matches
-    ):
+    for i in range(len(D)):
+        if (
+            D[candidate_idx] > atol + max_distance
+            or D[candidate_idx] == np.inf
+            or len(matches) == max_matches
```

Comment:
For safety, I recommend `len(matches) >= max_matches` (i.e., maybe somebody changes this function in the future and accidentally appends to `matches` twice each round and causes `len(matches)` to never be equal to `max_matches`.

Reply:
I don't know if `D[candidate_idx] == np.inf` is better or if `~np.isfinite(D[candidate_idx])` is better?

Reply:
> For safety, I recommend `len(matches) >= max_matches` (i.e., maybe somebody changes this function in the future and accidentally appends to `matches` twice each round and causes `len(matches)` to never be equal to `max_matches`.

Great perspective! I can see your point! I will revise it.

Reply:
> I don't know if `D[candidate_idx] == np.inf` is better or if `~np.isfinite(D[candidate_idx])` is better?

Maybe this is just me...but the latter forces me to keep in mind the operator `~` and also think about the return value of `np.isfinite()`, and  then put them together.

Reply:
I agree. That is why the original code used `np.isfinite` (because it also detects `np.nan`) and, therefore, does more and is safer. Given this, I think we should go with `np.isfinite` approach